docs: improve config presets (#7796)
renovatebot/renovate: docs/usage/config-presets.md

## Goals of Preset Configs

The main reason for supporting preset configs is to decrease duplication.
By using a preset config you:

1. Avoid duplicating the same config across all your repositories
2. Can use someone else's configuration and extend it

Renovate's configuration is self-documenting, because you can fill in the `"description"` field in all preset configs.

## Implementation Approach

In order to achieve these goals, preset configs allow for a very modular approach - preset configs can be as small as a partial package rule or as large as an entire configuration, like an `eslint` config.

## Preset Hosting

Presets can be defined using either npm packages, or with GitHub/GitLab repositories.

> Bitbucket-hosted presets are not supported for now.

Reviewer: I would maybe say that something like "presets hosted elsewhere are not supported." Or maybe just remove this sentence entirely?

Author: ```suggestion
Presets that are hosted elsewhere are not supported.
```

- `gitlab>abc/foo`: gitlab repository `https://gitlab.com/abc/foo` and preset `default`
- `gitlab>abc/foo:xyz`: gitlab repository `https://gitlab.com/abc/foo` and preset `xyz`

> It's easier to use GitHub/GitLab based preset hosting, as you avoid the "publish" step that is necessary with a npm package.

Author: Do we want to change this ordered list into a table as well?

Can we maybe copy paste the table from one of my other PRs? (I think the bottom table in the file for PR #7722 might be suitable???)
